core/src/main/java/com/backblaze/b2/json/B2JsonUnionBaseHandler.java

```diff
@@ -93,7 +101,7 @@
             if (!hasSuperclass(typeClass, clazz)) { // use clazz.isAssignableFrom(typeClass)?
                 throw new B2JsonException(typeClass + " is not a subclass of " + clazz);
             }
-            final B2JsonTypeHandler<?> handler = handlerMap.getHandler(typeClass);
+            final B2JsonTypeHandler<?> handler = b2JsonHandlerMap.getHandler(typeClass);
```

Comment:
why getHandler() instead of getUninitializedHandler()?  if there's a reason, maybe add a comment, since by default i would expect to only see getUninitializedHandler() calls during initializeImplementation().  ...or maybe i'm just confused.  :)

Reply:
Good question.  Will fix.

Reply:
thx.